Zend/zend_opcode.c

```diff
@@ -367,12 +367,13 @@ ZEND_API void destroy_op_array(zend_op_array *op_array)
 		zend_llist_apply_with_argument(&zend_extensions, (llist_apply_with_arg_func_t) zend_extension_op_array_dtor_handler, op_array);
 	}
 	if (op_array->arg_info) {
-		uint32_t num_args = op_array->num_args;
+		int32_t num_args = op_array->num_args;
+		int32_t i = op_array->fn_flags & ZEND_ACC_HAS_RETURN_TYPE ? -1 : 0;
 
 		if (op_array->fn_flags & ZEND_ACC_VARIADIC) {
 			num_args++;
 		}
-		for (i = 0; i < num_args; i++) {
+		for ( ; i < num_args; i++) {
 			zend_string_release(op_array->arg_info[i].name);
```

Comment:
If this line is commented out, we don't have a segfault:

http://chat.stackoverflow.com/transcript/message/21204932#21204932

I can also confirm on my machine.

Checking for NULL like this fixes segfault:

    if (op_array->arg_info[i].name) {
        zend_string_release(op_array->arg_info[i].name);
    }

However, I now get a memleak:

    [Wed Jan 28 03:02:11 2015]  Script:  '-'
    /Users/ajf/Projects/2014/PHP/php-src/Zend/zend_compile.c(0) :  Freeing 0x00860020 (8781824 bytes), script=-
    /Users/ajf/Projects/2014/PHP/php-src/Zend/zend_alloc.c(2247) : Actual location (location was relayed)
    === Total 1 memory leaks detected ===

:(